index.html

```diff
@@ -1,882 +1,1046 @@
-                        break;
-                    default:
-                        var error = JSON.stringify([4, receivedMessageId]);
-                        _websocket.send(error);
-                        break;
+                $("#transactions").html("");
+
+                for (
+                  var i = 0;
+                  i <
+                  JSON.parse(sessionStorage.getItem("TransactionId")).length;
+                  i++
+                ) {
+                  if (
+                    JSON.parse(sessionStorage.getItem("TransactionId"))[i] ==
+                    null
+                  ) {
+                    continue;
+                  }
+                  $("#transactions").append(
+                    JSON.parse(sessionStorage.getItem("TransactionId"))[i] +
+                      "<br>"
+                  );
                 }
-
+                $("#TransToStp").val(TransactionId);
+                //logMsg(TransactionId);
+              }
+              logMsg("Response received successfully!");
+
+              logMsg(JSON.stringify(ddata));
+            } else if (JSON.parse(msg.data)[0] === 4) {
+              logMsg("Data exchange failed - JSON is not accepted!");
+            } else if (JSON.parse(msg.data)[0] === 2) {
+              // Request from CSMS to CS
+
+              var receivedMessageId = ddata[1];
+              var receivedMessageAction = ddata[2];
+              var receivedMessagePayload = ddata[3];
+
+              logMsg(msg.data); // log message data
+
+              switch (ddata[2]) {
+                case "Reset":
+                  //Reset type SOFT, HARD
+                  var ResetS = JSON.stringify([
+                    3,
+                    receivedMessageId,
+                    {
+                      status: "Accepted",
+                    },
+                  ]);
+                  _websocket.send(ResetS);
+                  location.reload();
+                  break;
+                case "RemoteStopTransaction":
+                  //TransactionID
+                  var remStp = JSON.stringify([
+                    3,
+                    receivedMessageId,
+                    {
+                      status: "Accepted",
+                    },
+                  ]);
+                  _websocket.send(remStp);
+                  var stop_id = JSON.parse(msg.data)[3].transactionId;
+                  stopTransaction(stop_id);
+                  $(".indicator").hide();
+                  $("#yellow").show();
+                  break;
+                case "RemoteStartTransaction":
+                  //Need idTag, connectorId (map - ddata[3])
+                  var remStrt = JSON.stringify([
+                    3,
+                    receivedMessageId,
+                    {
+                      status: "Accepted",
+                    },
+                  ]);
+                  _websocket.send(remStrt);
+                  $("#idTag").val(ddata[3].idTag);

```

Comment:
This sets the idTag in the simulator to what was sent by the CSMS as central remote token.